yt/sample_data/api.py

```diff
@@ -0,0 +1,182 @@
+"""
+This is a collection of helper functions to yt.load_sample
+"""
+import json
+import re
+from functools import lru_cache
+from itertools import chain
+from pathlib import Path
+from typing import Optional, Union
+
+import pkg_resources
+
+from yt.funcs import mylog
+from yt.utilities.on_demand_imports import (
+    _pandas as pd,
+    _pooch as pooch,
+    _requests as requests,
+)
+
+num_exp = re.compile(r"\d*(\.\d*)?")
+byte_unit_exp = re.compile(r"[KMG]?B")
+
+
+def _parse_byte_size(s: str):
+    """
+    Convert a string size specification to integer byte size.
+
+    This function should be insensitive to case and whitespace.
+    Examples
+    --------
+    >>> _parse_byte_size("11B")
+    11
+    >>> _parse_byte_size("10 kB")
+    10000
+    >>> _parse_byte_size("67.4GB")
+    67400000000
+    >>> _parse_byte_size("0.1 MB")
+    100000
+    >>> _parse_byte_size(".1 MB")
+    100000
+    >>> _parse_byte_size(None)
+    <NA>
+    >>> from numpy import nan
+    >>> _parse_byte_size(nan)
+    <NA>
+    """
+    try:
+        s = s.upper()
+    except AttributeError:
+        # input is not a string (likely a np.nan)
+        return pd.NA
+    val = float(re.search(num_exp, s).group())
+    unit = re.search(byte_unit_exp, s).group()
+    exponent = {"B": 0, "KB": 3, "MB": 6, "GB": 9}[unit]
```

Comment:
Genuinely surprised this isn't in the stdlib!  Also I always forget the meaning of when an `i` is in the name like `MiB` but I'll assume this is correct.

Reply:
1) this may be in the stdlib, I don't remember checking this... I will update this thread after researching it more.
2) I don't think any existing value in this column is using an "i"

Reply:
I don't think it is, I just was noting that I always get confused.

Reply:
I couldn't find anything to support this in the stdlib... but what I DID find was a SO post https://stackoverflow.com/questions/12523586/python-format-size-application-converting-b-to-kb-mb-gb-tb

In fact this reminds me of an early concern from @cphyc , that I failed to address (until now), regarding the fact that 1KB = 1024B, rather than 1KB=1000B as I previously assumed.
Implementing the correct version of this parser made me realise that file sizes are often given with a limited amount of digits, which make them intrinsically fuzzy because, for instance "4.66GB" doesn't translate to an integer (interpreting `"GB"` as `*1024**3`).
Please, could one or either of you review this bit again ?